src/base/baseContractFactory.ts

```diff
@@ -0,0 +1,14 @@
+export abstract class baseContractFactory {
+  protected readonly _chainName: string;
+
+  protected constructor(_chainName: string) {
```

Comment:
For `_chainName` instead of using string can we either use an enum of supported chains or @bitgo/statics Coin object?

Reply:
Hi mark, we actually don't need this _chainName, was added for a first implementation and we did not use it anymore
We are deleting this property. Ty

we handle supported coins with this object
[src/index.ts#L11](https://github.com/BitGo/eth-contracts/blob/STLX-5-generic-library/src/index.ts#L11)

And then here we check for supported coins
[src/index.ts#L28](https://github.com/BitGo/eth-contracts/blob/STLX-5-generic-library/src/index.ts#L28)